drf_haystack/serializers.py

```diff
@@ -131,24 +138,69 @@ def get_fields(self):
                     warnings.warn("Field '{field}' already exists in the field list. This *will* "
                                   "overwrite existing field '{field}'".format(field=field_name))
                 field_mapping[field_name] = declared_fields[field_name]
-
         return field_mapping
 
     def to_representation(self, instance):
         """
+        If we have a serializer mapping, use that.  Otherwise, use standard serializer behavior
+        """
+        serializers = getattr(self.Meta, "serializers", None)
+        if not serializers:
+            return self.default_to_representation(instance)
+
+        index = instance.searchindex
+        serializer_class = serializers.get(type(index), None)
+        if not serializer_class:
+            raise ImproperlyConfigured("Could not find serializer for %s in mapping" % index)
+        return serializer_class(context=self._context).to_representation(instance)
+
+    def default_to_representation(self, instance):
```

Comment:
I think my intention was a modular composition that would allow downstream developers the flexibility to modify just the parts they wanted to when subclassing.  But now that I think about it, a cleaner solution for that might be something like (pseudo-code):

```python
def to_representation(self, instance):
    if we_have_serializer_mapping:
        return self.multi_serializer_representation(instance)
    return self.regular_representation(instance)
```

As one who's used many open-source projects and often been frustrated by inflexibility, I'd really like to separate pieces of functionality like this so another developer could just change the mapping behaviour or just the regular behavior if they wanted to.

If you have suggestions for a better name than `regular_representation`...

As for the `highlighted` fields...yeah, I haven't actually ever even looked at the highlighter code.  I'm just scanning `HighlighterMixin` now...okay, I'll have to take a look at it.